### PR TITLE
Adds INavbarProps.stickyToTop

### LINKED ISSUE #3055: Add stickyToTop prop to Navbar
#### Environment

- __Package version(s)__: ^3.7.0
- __Browser and OS versions__: Chrome 69, Fedora 27

#### Feature request

Add `stickyToTop` or similar named prop to Navbar, similar to `fixedToTop` prop.

`fixedToTop` uses `position: fixed` to affix Navbar to top of screen. Per W3C this is supposed to attach to the viewport and not any other elements on the page: https://www.w3.org/TR/css-position-3/#valdef-position-fixed

This means any sidebar element will be below the Topbar, as the Topbar stretches across the full width of the screen when fixed:

image

The `position: sticky` property, however, can be attached to an element and not just the viewport: https://www.w3.org/TR/css-position-3/#valdef-position-sticky

This allows the stickied element to respect its first parent with `position: relative`:

image

image

`position: sticky` has been in Chrome since 2012: https://developers.google.com/web/updates/2012/08/Stick-your-landings-position-sticky-lands-in-WebKit

All non-mobile browsers have support: https://caniuse.com/#feat=css-sticky

### PR DESCRIPTION
**Pull request by jtreminio**
#### Checklist

- [ ] [Enable CircleCI for your fork](https://circleci.com/add-projects)
- [ ] Include tests
- [x] Update documentation

### CODE REVIEW

**Comment by palantirtech** [comment]
Thanks for your interest in [palantir/blueprint](https://github.com/palantir/blueprint), @jtreminio! Before we can accept your pull request, you need to sign our contributor license agreement - just visit https://cla.palantir.com/ and follow the instructions. Once you sign, I'll automatically update this pull request.

**Comment by jtreminio** [comment]
circleci currently failing on https://github.com/palantir/blueprint/pull/3056/files#diff-15f5f08e34c8f3c6375626e31dbe6819R41

I'm unsure how you want those lines formatted, though. Input would be appreciated.

File: packages/core/src/components/navbar/_navbar.scss [hunk 1]
```diff
@@ -56,12 +56,19 @@ $dark-navbar-background-color: $dark-gray5 !default;
...
+      right: 0;
+      left: 0;
+    }
+
+    &.#{$ns}-sticky-top {
+      position: sticky;
+      top: 0;
+      right: 0;
+      left: 0;
+    }
```
Comment (giladgray) [inline packages/core/src/components/navbar/_navbar.scss]: indentation is wrong here. too many spaces.

File: packages/core/src/components/navbar/navbar.md [hunk 2]
```diff
@@ -10,7 +10,11 @@ Enable the `fixedToTop` prop to attach a navbar to the top of the viewport using
 `position: fixed; top: 0;`. This is so-called "sticky" behavior: the navbar
 stays at the top of the screen as the user scrolls through the document.
```
Comment (giladgray) [inline packages/core/src/components/navbar/navbar.md]: this sentence is no longer correct if the component offers actual `sticky` support.

File: packages/core/src/components/navbar/navbar.tsx [hunk 3]
```diff
@@ -20,6 +20,11 @@ export interface INavbarProps extends IProps, HTMLDivProps {
      * Whether this navbar should be fixed to the top of the viewport (using CSS `position: fixed`).
      */
     fixedToTop?: boolean;
+
+    /**
+     * Whether this navbar should be fixed to the top of the viewport (using CSS `position: sticky`).
```
Comment (giladgray) [inline packages/core/src/components/navbar/navbar.tsx]: use a better word than "fixed" to describe this behavior

Comment (giladgray) [reply packages/core/src/components/navbar/navbar.tsx]: also please add a note that this feature is unsupported on IE11.

File: packages/core/src/components/navbar/navbar.md [hunk 4]
```diff
@@ -10,7 +10,11 @@ Enable the `fixedToTop` prop to attach a navbar to the top of the viewport using
...
 
-This modifier is not illustrated here because it breaks the document flow.
+Enable the `stickyToTop` prop to attach a navbar to the top of the viewport using
+`position: sticky; top: 0;`. Similar to `fixedToTop` but attaches to parent element
+vs the viewport.
```
Comment (giladgray) [inline packages/core/src/components/navbar/navbar.md]: add a note that this feature is not supported on IE11.

**Comment by jtreminio** [comment]
Documentation updated to reflect the fact that the navbar attaches itself to its _parent_ and scrolls up and down with the document until the navbar reaches the top of the viewport, where it attaches itself.

File: packages/core/src/components/navbar/navbar.tsx [hunk 5]
```diff
@@ -38,10 +38,14 @@ export class Navbar extends React.PureComponent<INavbarProps, {}> {
...
         const { children, className, fixedToTop, stickyToTop, ...htmlProps } = this.props;
-        const classes = classNames(Classes.NAVBAR, {
-            [Classes.FIXED_TOP]: fixedToTop,
-            [Classes.FIXED_TOP]: stickyToTop
-        }, className);
+        const classes = classNames(
+            Classes.NAVBAR,
+            {
+                [Classes.FIXED_TOP]: fixedToTop,
+                [Classes.FIXED_TOP]: stickyToTop,
```
Comment (giladgray) [inline packages/core/src/components/navbar/navbar.tsx]: oops same class. did this ever work?

File: packages/core/src/components/navbar/navbar.md [hunk 6]
```diff
@@ -7,12 +7,13 @@ Navbars present useful navigation controls at the top of an application.
...
+`position: fixed; top: 0;`. The navbar stays at the top of the screen and above
+other elements as the user scrolls through the document.
 
-Enable the `stickyToTop` prop to attach a navbar to the top of the viewport using
-`position: sticky; top: 0;`. Similar to `fixedToTop` but attaches to parent element
-vs the viewport.
+Enable the `stickyToTop` prop to attach a navbar to the top of its parent container using
+`position: sticky; top: 0;`. If navbar is placed below the top of the viewport, it will
+move as the user scrolls through the document, attaching itself to the top of the viewport
+once it is reached. IE11 does not support this feature.
```
Comment (giladgray) [inline packages/core/src/components/navbar/navbar.md]: Please bold the line about IE11.

File: packages/core/src/components/navbar/navbar.tsx [hunk 7]
```diff
@@ -22,7 +22,7 @@ export interface INavbarProps extends IProps, HTMLDivProps {
     fixedToTop?: boolean;
 
     /**
-     * Whether this navbar should be fixed to the top of the viewport (using CSS `position: sticky`).
+     * Whether this navbar should be attached to the top of its parent container (using CSS `position: sticky`).
```
Comment (giladgray) [inline packages/core/src/components/navbar/navbar.tsx]: also add note about lack of IE11 support here.

**Comment by giladgray** [comment]
@jtreminio any update here? if you merge the latest `develop` then your build should pass.

**Comment by jtreminio** [comment]
Apologies for lack of action on my part. Applied requested changes.

**Comment by giladgray** [comment]
@jtreminio after some more thought and internal discussion, we have decided that this feature is not worth supporting in Blueprint itself for two reasons: 

1. it does not match our browser support commitments (no IE11) and 
1. it is trivially implemented as needed with nothing more than a single CSS override: 
  `position: fixed -> sticky;`.

thanks for taking the time to do the work and respond to my reviews! gonna close this out now.

**Pull request closed without merging**